### ISSUE TITLE
Synchronization and atomics chapter desing

### ISSUE THREAD

**Issue by PetroBondar** [KhronosGroup/SYCL_Reference#124]
Currently, the "Atomic Operations" chapter looks like this:
![image](https://github.com/KhronosGroup/SYCL_Reference/assets/122783909/632e6164-82dc-43b9-a5b7-0bcfd31efafb)

We propose to rename this chapter to "Synchronization and atomics" and implement the following structure:

- Synchronization and atomics
  - Barriers and fences ([4.15.1. Barriers and fences](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:barriers-fences))
   - ``sycl::device_event ``  ([4.15.2. device_event class](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:barriers-fences))
   - Atomic references ([4.15.3. Atomic references](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:atomic-references))
   - Interaction with host code ([4.15.5. Interaction with host code](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_interaction_with_host_code))

Also, should we include deprecated [Atomic types](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:atom-types-depr) in the reference guide?

We will create the PR with updates soon. Feel free to comment if you have any suggestions on the structure.

**Comment by PetroBondar** [comment]
Created PR: https://github.com/KhronosGroup/SYCL_Reference/pull/126

**Comment by mkinsner** [comment]
> Also, should we include deprecated [Atomic types](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:atom-types-depr) in the reference guide?

Yes, because the deprecated features are still in the specification and need to be supported, they should be in the reference guide.
